cypress-io/cypress
# chore: add optimized deps to reduce CI flake for CT tests

### User facing changelog
na

### Additional details
Adding additional optimizedDeps to reduce flake in CI. These deps can cause vite to do a full page load in the middle of a test causing it to fail.

I had to add some deps that aren't clear why they are being included, such as `@cypress/parse-domain`. I based these deps off of the failures from [this build](https://app.circleci.com/pipelines/github/cypress-io/cypress/40030/workflows/3c33d2cf-6453-40f9-b38a-09db24d2b2d2/jobs/1642181/parallel-runs/1?filterBy=ALL) and some local testing.


### Steps to test
Run `yarn workspace @packages/app cypress:run:ct`, notice no vite dep detection logs.

### How has the user experience changed?
na

### PR Tasks


- [na] Have tests been added/updated?
- [na] Has the original issue (or this PR, if no issue exists) been tagged with a release in ZenHub? (user-facing changes only)
- [na] Has a PR for user-facing changes been opened in [`cypress-documentation`](https://github.com/cypress-io/cypress-documentation)? 
- [na] Have API changes been updated in the [`type definitions`](https://github.com/cypress-io/cypress/blob/develop/cli/types/cypress.d.ts)?
